ModelBundle/Repository/SiteRepository.php

```diff
@@ -118,13 +118,17 @@ public function findByAliasDomain($domain)
             )
         ));
 
+        $ids = array();
         if (is_array($commandResult) && array_key_exists('ok', $commandResult) && $commandResult['ok'] == 1) {
             foreach ($commandResult['results'] as $siteId) {
-                return $this->findOneBySiteId($siteId['_id']);
+                $ids[] = $siteId['_id'];
```

Comment:
I understand what you're proposing but I'm not a big fan of it for two related reasons:

1. the mecanism to determinate the correct alias is complex. It requires to find the largest matching pattern among language prefixes once the domains are found. Doing this here inject a lot of business logic in the repo. The correct location for the business logic is on the php stuff not on the repo. In the repo you should only see simple data recovery, with simple filters.

2. We could consider doing this in Mongo thanks to a javascript function, but it will be very hard to develop a version for other database systems, such as mysql.